src/smtsolvers/ParallelScatterSplitter.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022, Antti Hyvarinen <antti.hyvarinen@gmail.com>
+ * Copyright (c) 2022, Seyedmasoud Asadzadeh <seyedmasoud.asadzadeh@usi.ch>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#ifndef OPENSMT_PARALLELSCATTERSPLITTER_H
+#define OPENSMT_PARALLELSCATTERSPLITTER_H
+
+#include "ScatterSplitter.h"
+
+#include <PTPLib/net/Channel.hpp>
+#include <PTPLib/common/Exception.hpp>
+#include <PTPLib/common/Printer.hpp>
+
+class ParallelScatterSplitter : public ScatterSplitter  {
+public:
+    ParallelScatterSplitter(SMTConfig & c, THandler & t);
+    void set_channel(PTPLib::net::Channel & ch) { channel = &ch; }   //SMTS Client owns the channel and should dirctely set the channel in ParallelScatterSpitter
+    PTPLib::net::Channel & getChannel() const { return *channel;};
+    void set_syncedStream(PTPLib::common::synced_stream & ss) { syncedStream = &ss; }  //SMTS Client owns the SyncedStream and should dirctely set the SyncedStream in ParallelScatterSpitter
+    vec<opensmt::pair<int,int>> const &  get_solver_branch()  const  { return solverBranch; }
+    void set_solver_branch(std::string solver_branch);
+    void mapSolverBranchToFrameId(uint32_t fid, vec<opensmt::pair<int,int>> && solverAddress) {
+        frameId_solverBranch[fid] = std::move(solverAddress);
+    }
+    void addBranchToFrameId(opensmt::span<opensmt::pair<int, int> const> && solver_branch, uint32_t fid);
+
+    inline vec<opensmt::pair<int,int>> const & get_solverBranch(Var v) { return frameId_solverBranch[theory_handler.getTMap().get_FrameId(v)]; }
+
+private:
+    PTPLib::net::Channel *  channel;
+    int                     trail_sent;
+    bool                    firstPropagation;
+    int                     numTriviallyPropagatedOnDl0;
+
+    vec<opensmt::pair<int,int>> solverBranch;
+    PTPLib::common::synced_stream * syncedStream;
+    typedef std::map<uint32_t, vec<opensmt::pair<int,int>>> map_frameId_solverBranch;
+    map_frameId_solverBranch frameId_solverBranch;
+    typedef std::map<Var ,uint32_t> map_var_frameId;
+    map_var_frameId var_frameId;
+
+    void shallLearnClauses () override;                                       // Check if solver is in clause share mode or single-query mode and then starts clause learning operation
+
+    void notifyFoundedResult(lbool const & result) const;                     // OpenSMT signalling to communication channel that it founded a result and will stop
```

Comment:
```suggestion
    void notifyFoundResult(lbool const & result) const;                     // OpenSMT signalling to communication channel that it found a result and will stop
```

Reply:
TFR, Sounds more natural to me with 'ed' ;)

Reply:
From dictionary:
verb: find; 3rd person present: finds; past tense: found; past participle: found
verb: found; 3rd person present: founds; past tense: founded; past participle: founded

I think you are mixing the verbs `find` and `found` :)
BTW, what is TFR?

Reply:
TFR = Thanks for reminding
;) = wink